src/types.rs

```diff
@@ -504,3 +505,11 @@ pub enum CharacterCodes {
     Tab = 0x09,         // \t
     VerticalTab = 0x0B, // \v
 }
+
+#[wasm_bindgen]
+pub struct SourceFileLike {
+    #[wasm_bindgen(readonly)]
+    pub text: &'static str,
+    #[wasm_bindgen(readonly, js_name = "lineMap")]
+    pub line_map: Option<Vec<usize>>,
```

Comment:
It is impossible to use this in `wasm_bindgen`. Serializing won't work either (without additional Javascript at least) since it needs to be modified in place.